lib/ABI/Definition.cpp

```diff
@@ -366,40 +366,53 @@ naturalAlignment(const abi::Definition &ABI,
   rc_return rc_recur naturalAlignment(ABI, *QT.UnqualifiedType().get(), Cache);
 }
 
+template<typename T>
+void abortIfFalse(bool Success, const T &Type) {
```

Comment:
`s/abortIfFalse/assertOnType/`?

Reply:
I went with `checkIfComputationFailed`, but also pulled the `Value == 0` check in as well.

Please take a look at the updated `ABI: make alignment-computation fails stricter` commit.

Reply:
If a function aborts unconditionally, it should contain "abort" in the name.
If a function aborts depending on a `bool` argument, it should contain "assert" in the name.

Reply:
To be fair, this is not really a function, as in, it's not a part of any interface. The only reason it exists is to unload repetitive checks from the functions that come next.

Either way, it looks like this now:
```cpp
template<typename T>
bool checkIfComputationFailed(std::optional<AlignmentInfo> ComputationResult,
                              const T &ThingToDumpOnFailure) {
  if (!ComputationResult) {
    std::string Error = "Unable to compute the alignment of "
                        + serializeToString(ThingToDumpOnFailure);
    revng_abort(Error.c_str());
  }

  return ComputationResult->Value == 0;
}

// users
  std::optional<AlignmentInfo> Result = naturalAlignment(*this, QT, Cache);
  if (checkIfComputationFailed(Result, QT))
    return std::nullopt;
```

What do you think the name should be?

Reply:
It's a bit weird how the check is organized. I'd make "can't fail" version of `naturalAlignment` and then instead of

```c++
  if (checkIfComputationFailed(Result, QT))
    return std::nullopt;
```

I'd put `if (xxx == 0) return std::nullopt;`.

Reply:
> I'd make "can't fail" version of `naturalAlignment`

I thought about it, but frankly, we already have way too many versions. I'd rather we avoid adding any more.

All I want is for all these practically identical bodies to be as short as possible so that they are at least somewhat reasonable to deal with.

To be honest, I this 🤏 close to just saying "screw it" and making a heavily templated `alignmentPITA_IMPL`-like function that turns all of these into one-liners like `auto whatever(/* args */) { return alignmentPITA_IMPL</* params */>(/* args */); }`

Reply:
```c++
  AlignmentInfo Result = myCantFail(*this, naturalAlignment(*this, QT, Cache));
  if (ComputationResult->Value == 0)
    return std::nullopt;
```

Reply:
Done (`assertOnFailure`).